app/assets/javascripts/modules/mobile-filters-modal.js

```diff
@@ -2,104 +2,57 @@ window.GOVUK = window.GOVUK || {}
 window.GOVUK.Modules = window.GOVUK.Modules || {};
 
 (function (Modules) {
-  function MobileFiltersModal ($module) {
-    this.$module = $module
-    this.$facetsBox = this.$module.querySelector('.facets__box')
-    this.$closeTriggers = this.$module.querySelectorAll('.js-close-filters')
-    this.$showResultsButton = this.$module.querySelector('.js-show-results')
-    this.$clearFiltersTrigger = this.$module.querySelector('.js-clear-selected-filters')
-    this.$body = document.querySelector('body')
-
-    this.$module.open = this.handleOpen.bind(this)
-    this.$module.close = this.handleClose.bind(this)
-    this.$module.clearFilters = this.handleClearFilters.bind(this)
-    this.$module.ModalFocus = this.handleModalFocus.bind(this)
-    this.$module.boundKeyDown = this.handleKeyDown.bind(this)
+  function MobileFiltersModal (module) {
+    this.module = module
+    this.triggerElement = document.querySelector('[data-toggle="mobile-filters-modal"][data-target="' + this.module.id + '"]')
+    this.clearFiltersTrigger = this.module.querySelector('.js-clear-selected-filters')
+    this.module.toggle = this.handleToggle.bind(this)
+    this.module.open = this.handleOpen.bind(this)
+    this.module.close = this.handleClose.bind(this)
+    this.module.clearFilters = this.handleClearFilters.bind(this)
   }
 
   MobileFiltersModal.prototype.init = function () {
-    var $triggerElement = document.querySelector(
-      '[data-toggle="mobile-filters-modal"][data-target="' + this.$module.id + '"]'
-    )
-    if ($triggerElement) {
-      $triggerElement.addEventListener('click', this.$module.open)
+    if (this.triggerElement) {
+      this.triggerElement.addEventListener('click', this.module.toggle)
+      this.triggerElement.setAttribute('aria-controls', this.module.id)
+      this.triggerElement.setAttribute('aria-expanded', 'false')
     }
 
-    if (this.$closeTriggers) {
-      for (var t = 0; t < this.$closeTriggers.length; t++) {
-        this.$closeTriggers[t].addEventListener('click', this.$module.close)
-      }
+    if (this.clearFiltersTrigger) {
+      this.clearFiltersTrigger.addEventListener('click', this.module.clearFilters)
     }
+  }
 
-    if (this.$clearFiltersTrigger) {
-      this.$clearFiltersTrigger.addEventListener('click', this.$module.clearFilters)
+  MobileFiltersModal.prototype.handleToggle = function (event) {
+    if (event) {
```

Comment:
Not sure this check needs to be in place? Is there a situation where `event` wouldn't be defined?

Reply:
@andysellick I believe it's because the tests call the functions without firing an event, so you wouldn't be able to call `.preventDefault()` on something that's undefined. Is this OK?